stream/pubsub.h

```diff
@@ -335,6 +340,89 @@ class PubSubHTTPEndpointImpl : public AbstractSubscriberObject {
     return result;
   }
 
+  ss::EntryResponse operator()(const std::string& entry_json, uint64_t current_index, idxts_t last) {
+    const ss::EntryResponse result = [&, this]() {
+      if (time_to_terminate_) {
+        return ss::EntryResponse::Done;
+      }
+      auto current_us = std::chrono::microseconds(0);
+      // Obtain current timestamp only when it's necessary by parsing the `entry_json`.
+      const auto GetCurrentUs = [this, &current_us, &entry_json]() -> std::chrono::microseconds {
+        if (!current_us.count()) {
+          current_us = ParseJSON<idxts_t>(entry_json.substr(0, entry_json.find('\t'))).us;
+        }
+        return current_us;
+      };
+      if (!serving_) {
+        if (current_index >= params_.i &&                                            // Respect `i`.
+            (params_.tail == 0u || (last.index - current_index) < params_.tail) &&   // Respect `tail`.
+            (from_timestamp_.count() == 0u || GetCurrentUs() >= from_timestamp_)) {  // Respect `since` and `recent`.
+          serving_ = true;
+        }
+        // Reached the end, didn't started serving and should not wait.
+        if (!serving_ && current_index == last.index && params_.no_wait) {
```

Comment:
And `>=` for index looks scary to me. Shouldn't it always increase by just one?

At the same time, `current_index == last.index ` looks too strict, what if we started from past the end for some reason?